### PR TITLE
Update workflows

### PR DESCRIPTION
**Pull request by bcantoni**
This brings over 2 recent workflow changes:
- add Python 3.10 released version
- don't fail the build if Slack notification fails for some reason

### CODE REVIEW

**Comment by bcantoni** [comment]
@jorwoods or @t8y8 want to double-check me on this? Once merged over to master branch, the Slack changes here should unblock jorwoods PRs from today after a pipeline re-run.

**Comment by jorwoods** [comment]
@bcantoni Looks great to me!

**Review by jorwoods — APPROVED** [review]

**Review by ashwinar — APPROVED** [review]

**Pull request merged**
